RELEASE_NOTES.md

```diff
@@ -7,6 +7,11 @@ This file contains a high-level description of this package's evolution. Release
 
 ## 4.7.3 - TBD
 
+* [Bug Fix] Library was modified so that rewriting the same attribute happens without deleting the attribute, to avoid a limit on how many times this may be done in HDF5. This fix was thought to be in 3.6.2 but was not. See [https://github.com/Unidata/netcdf-c/issues/350].
```

Comment:
@WardF Please note that this entry is not correct. It should say 4.6.2, not 3.6.2.